karma.conf.js

```diff
@@ -0,0 +1,109 @@
+// Karma configuration
+
+module.exports = function(config) {
+  config.set({
+
+    // base path that will be used to resolve all patterns (eg. files, exclude)
+    basePath: 'ubcpi/static/js',
+
+
+    plugins: [
+      'karma-coverage',
+      'karma-jasmine',
+	  'karma-jasmine-jquery',
+      'karma-chrome-launcher',
+      'karma-firefox-launcher',
+      'karma-phantomjs-launcher',
+      'karma-ng-html2js-preprocessor'
+    ],
+
+    // frameworks to use
+    // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
+    frameworks: [ 'jasmine-jquery', 'jasmine' ],
+
+
+    // list of files / patterns to load in the browser
+    files: [
+      '../../../node_modules/jquery/tmp/jquery.js',
+      '../../../node_modules/angular/angular.js',
+      '../../../node_modules/angular-messages/angular-messages.js',
+      '../../../node_modules/angular-sanitize/angular-sanitize.js',
+      '../../../node_modules/angular-cookies/angular-cookies.js',
+      '../../../node_modules/d3/d3.js',
+      '../../../node_modules/angular-mocks/angular-mocks.js',
+      '../../../node_modules/jasmine-core/lib/jasmine-core/jasmine.js',
```

Comment:
@sarina they are only used in tests (karma.conf.js is the test runner set up). The instruction on how to run tests are in the readme.md file. When you run make deps, it will install those external dependencies. They are node modules coming from npmjs.